x/incentives/client/cli/tx.go

```diff
@@ -128,3 +132,136 @@ func NewAddToGaugeCmd() *cobra.Command {
+func NewCmdHandleCreateGroupsProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-groups-proposal [pool-id-pairs] [flags]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a create groups proposal",
+		Long: strings.TrimSpace(`Submit a create groups proposal.
+
+Passing in pool-id-pairs separated by commas would be parsed automatically to a single set for a single group.
+If a semicolon is presented, that would be parsed as pool IDs for separate group.
+Don't forget the single quotes around the pool IDs!
+Ex) create-groups-proposal '1,2;3,4,5;6,7' ->
+Group 1: Pool IDs 1, 2
+Group 2: Pool IDs 3, 4, 5
+Group 3: Pool IDs 6, 7
+
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			content, err := parseCreateGroupArgToContent(cmd, args[0])
+			if err != nil {
+				return err
+			}
+
+			from := clientCtx.GetFromAddress()
+
+			depositStr, err := cmd.Flags().GetString(govcli.FlagDeposit)
+			if err != nil {
+				return err
+			}
+			deposit, err := sdk.ParseCoinsNormalized(depositStr)
+			if err != nil {
+				return err
+			}
+
+			msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+			if err != nil {
+				return err
+			}
+
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+	cmd.Flags().String(govcli.FlagTitle, "", "title of proposal")
+	cmd.Flags().String(govcli.FlagDescription, "", "description of proposal")
+	cmd.Flags().String(govcli.FlagDeposit, "", "deposit of proposal")
+	cmd.Flags().Bool(govcli.FlagIsExpedited, false, "If true, makes the proposal an expedited one")
+	cmd.Flags().String(govcli.FlagProposal, "", "Proposal file path (if this path is given, other proposal flags are ignored)")
+
+	return cmd
+}
+
+func parseCreateGroupArgToContent(cmd *cobra.Command, arg string) (govtypes.Content, error) {
+	title, err := cmd.Flags().GetString(govcli.FlagTitle)
+	if err != nil {
+		return nil, err
+	}
+
+	description, err := cmd.Flags().GetString(govcli.FlagDescription)
+	if err != nil {
+		return nil, err
+	}
+
+	createGroupRecords, err := ParseCreateGroupRecords(arg)
+	if err != nil {
+		return nil, err
+	}
+
+	content := &types.CreateGroupsProposal{
+		Title:        title,
+		Description:  description,
+		CreateGroups: createGroupRecords,
+	}
+
+	return content, nil
+}
+
+func ParseCreateGroupRecords(arg string) ([]types.CreateGroup, error) {
+	poolIds2DArray, err := parseStringTo2DArray(arg)
+	if err != nil {
+		return nil, err
+	}
+
+	createGroupRecords := []types.CreateGroup{}
+
+	for _, poolIds := range poolIds2DArray {
+		createGroupRecords = append(createGroupRecords, types.CreateGroup{
+			PoolIds: poolIds,
+		})
+	}
+
+	return createGroupRecords, nil
+}
+
+func parseStringTo2DArray(input string) ([][]uint64, error) {
```

Comment:
does it make sense to move this helper to `osmocli`?

Reply:
Makes sense, moved to osmocli here https://github.com/osmosis-labs/osmosis/pull/6547/commits/a48496d2db0e7d642ba0ad54832b70558d66ff7a